vowpalwabbit/core/include/vw/core/reductions/ftrl.h

```diff
@@ -3,8 +3,41 @@
 // license as described in the file LICENSE.
 
 #pragma once
+#include "vw/core/reductions/gd.h"
 #include "vw/core/vw_fwd.h"
 
+#include <cstddef>
+#include <cstdint>
+
+namespace
```

Comment:
Anonymous namespaces should not be used in headers. This should be in a namespace like `VW::reductions`